### PR TITLE
net/gnrc: Remove deprecated module gnrc_nettest

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This removes the `nettest` module, deprecated in #12594, and scheduled for removal for 2020.07 at the latest.

### Testing procedure
- Check that all references to the module are gone

### Issues/PRs references
Module deprecated in #12594

### CODE REVIEW

File: LOSTANDFOUND.md [hunk 1]
```diff
@@ -40,6 +40,14 @@ This way, their names are never removed from the RIOT repository.
 
 # Removed Features
 
+### sys/net/gnrc/nettest [c84cedb07212a90932e446f5d57391d1edd486d0]
+Author:
+- Martine Lenders <mlenders@inf.fu-berlin.de>
+
+Reasons for removal:
+- Not maintained anymore
+- Not used in GNRC
```
Comment (miri64) [inline LOSTANDFOUND.md:49]:
```suggestion
- Not used for testing GNRC
```

It was never intended to be used in GNRC but for testing GNRC modules.

**Comment by leandrolanzieri** [comment]
> Blocking, so it is not accidentally removed before the 2020.07 release.

As the doc says " It will be removed after the 2020.07 release _at the latest_." I thought it would happen now :) That's why I opened this for 2020.07. But that's OK.

**Comment by miri64** [comment]
You marked the wrong part ;-P "It will be removed _after_ the 2020.07 release at the latest". But maybe we should re-think that working "at the latest" can be interpreted to also delete this directly after deprecation ;-).

**Comment by miri64** [comment]
Please rebase, so GitHub Actions are included.

**Comment by leandrolanzieri** [comment]
Done

**Pull request merged**